src/CSharpFunctionalExtensions.FluentAssertions/ResultExtensions.cs

```diff
@@ -0,0 +1,51 @@
+﻿using FluentAssertions;
+using FluentAssertions.Execution;
+using FluentAssertions.Primitives;
+
+namespace CSharpFunctionalExtensions.FluentAssertions;
+
+public static class ResultExtensions
+{
+    public static ResultAssertions Should(this Result instance) => new(instance);
+}
+
+public class ResultAssertions : ReferenceTypeAssertions<Result, ResultAssertions>
+{
+    public ResultAssertions(Result instance) : base(instance) { }
+
+    protected override string Identifier => "Result";
+
+    /// <summary>
+    /// 
+    /// </summary>
+    /// <param name="because"></param>
+    /// <param name="becauseArgs"></param>
+    /// <returns></returns>
+    public AndConstraint<ResultAssertions> BeSuccessful(string because = "", params object[] becauseArgs)
+    {
+        Execute.Assertion
+            .BecauseOf(because, becauseArgs)
+            .Given(() => Subject.IsSuccess)
+            .ForCondition(isSuccess => isSuccess)
+            .FailWith("Expected Result to be successful but found error");
```

Comment:
found failure?

Reply:
Yah, that lines up better with the language used by the library.

Maybe...
> "Expected Result to be successful but it failed"

Reply:
Does `FailWith()` have access to the `Result` through an overload that accepts a delegate? Maybe you could call `.ToString()` on the `Result.Error` in the message...

Reply:
The expected -> found verbiage is from the FluentAssertions docs (see the second code snippet). I think it makes sense to use CSharpFunctionalExtensions language since it's the primary library in our package naming convention.
https://fluentassertions.com/extensibility/#building-your-own-extensions